File: services/eval/src/metrics/hit-rate.ts
export function hitRate(retrievedIds: string[], relevantIds: string[]): 0 | 1 {
  if (relevantIds.length === 0) return 0;
  const relevantSet = new Set(relevantIds);
  return retrievedIds.some((id) => relevantSet.has(id)) ? 1 : 0;
}

File: services/eval/src/rag_eval/metrics/hit_rate.py
from collections.abc import Sequence


def hit_rate(retrieved_ids: Sequence[str], relevant_ids: Sequence[str]) -> int:
    """1 if at least one relevant id was retrieved anywhere, else 0."""
    if len(relevant_ids) == 0:
        return 0
    relevant_set = set(relevant_ids)
    return 1 if any(chunk_id in relevant_set for chunk_id in retrieved_ids) else 0
